### PR TITLE
Fix #167: REST API documentation

### LINKED ISSUE #167: REST API documentation
Document the REST API and describe possible use cases.

### PR DESCRIPTION
**Pull request by romanstrobl**

### CODE REVIEW

File: docs/User-Data-Store-API.md [hunk 1]
```diff
@@ -0,0 +1,945 @@
...
+### Claims API
+
+- [GET /claims](#fetch-claims) - Fetch claims
+- [POST /admin/claims](#create-claims) - Create claims
+- [PUT /admin/claims](#update-claims) - Update claims
+- [DELETE /admin/claims](#delete-claims) - Delete claims
+
+### User Claims API (Deprecated)
+
+This REST API is deprecated, see Swagger for usage information.
```
Comment (banterCZ) [inline docs/User-Data-Store-API.md:47]: What about using a warning box?
```suggestion
<!-- begin box warning -->
This REST API is deprecated, see Swagger for usage information.
<!-- end -->
```

File: docs/User-Data-Store-API.md [hunk 2]
```diff
@@ -0,0 +1,945 @@
...
+
+Use HTTP Basic authentication to access the REST API, for example:
+
+Authorization: Basic YWRtaW46YWRtaW4=
+
+The username and password is a Base-64 encoded string in format username:password.
+
+The following paths require a ROLE_WRITE authority:
+- `/admin/**` - supported REST API for User Data Store
+- `/public/**` - deprecated REST API for user claims
```
Comment (banterCZ) [inline docs/User-Data-Store-API.md:95]:
```suggestion
- `/public/**` - **deprecated** REST API for user claims
```

File: docs/User-Data-Store-API.md [hunk 3]
```diff
@@ -0,0 +1,945 @@
...
+
+### Authentication
+
+Access to the REST API is authenticated. See chapter [Setting Up REST Service Credentials](Deploying-User-Data-Store.md#setting-up-rest-service-credentials) for configuring access to the REST API.
+
+Use HTTP Basic authentication to access the REST API, for example:
+
+Authorization: Basic YWRtaW46YWRtaW4=
+
+The username and password is a Base-64 encoded string in format username:password.
```
Comment (banterCZ) [inline docs/User-Data-Store-API.md:91]:
```suggestion
`Authorization: Basic YWRtaW46YWRtaW4=`

The username and password is a Base-64 encoded string in the format `username:password`.
```

**Review by banterCZ — APPROVED** [review]
Looks good to me.

File: docs/User-Data-Store-API.md [hunk 4]
```diff
@@ -0,0 +1,945 @@
...
+#### Request
+
+- Headers:
+    - `Authorization: Basic ...`
+
+##### Query Params
+
+| Param        | Type                                                     | Description                                                         |
+|--------------|----------------------------------------------------------|---------------------------------------------------------------------|
+| `userId`     | `String`<span class="required" title="Required">*</span> | User identifier of the owner of deleted documents.                  |
```
Comment (jnpsk) [inline docs/User-Data-Store-API.md:342]: As it is in the request, the documents are yet to be deleted. This past tense is used in more places.

```suggestion
| `userId`     | `String`<span class="required" title="Required">*</span> | User identifier of the owner of documents to be deleted.                  |
```

File: docs/User-Data-Store-API.md [hunk 5]
```diff
@@ -0,0 +1,945 @@
...
+##### Request Params
+
+| Parameter                                                        | Type     | Description                                                                          |
+|------------------------------------------------------------------|----------|--------------------------------------------------------------------------------------|
+| `userId`<span class="required" title="Required">*</span>         | `String` | User identifier of document owner.                                                   |
+| `documentId`<span class="required" title="Required">*</span>     | `String` | Identifier of the related document.                                                  |
+| `attachmentType`<span class="required" title="Required">*</span> | `String` | One of `text`, `image_base64`, `binary_base64`.                                      |
+| `attachmentData`<span class="required" title="Required">*</span> | `String` | Base-64 encoded attachment data.                                                     |
+| `externalId`                                                     | `String` | Optional external identifier of the photo (e.g. identifier used in the bank system). |
```
Comment (jnpsk) [inline docs/User-Data-Store-API.md:662]:
```suggestion
| `externalId`                                                     | `String` | Optional external identifier of the attachment (e.g. identifier used in the bank system). |
```

File: docs/User-Data-Store-API.md [hunk 6]
```diff
@@ -0,0 +1,945 @@
...
+}
+```
+
+##### Request Params
+
+| Parameter                                                        | Type     | Description                                                                          |
+|------------------------------------------------------------------|----------|--------------------------------------------------------------------------------------|
+| `attachmentType`<span class="required" title="Required">*</span> | `String` | One of `text`, `image_base64`, `binary_base64`.                                      |
+| `attachmentData`<span class="required" title="Required">*</span> | `String` | Base-64 encoded attachment data.                                                     |
+| `externalId`                                                     | `String` | Optional external identifier of the photo (e.g. identifier used in the bank system). |
```
Comment (jnpsk) [inline docs/User-Data-Store-API.md:723]:
```suggestion
| `externalId`                                                     | `String` | Optional external identifier of the attachment (e.g. identifier used in the bank system). |
```

**Review by jnpsk — APPROVED** [review]
Nice!
I found only a few minor problems. Those that should be fixed:
- There is inconsistency in the placing of the asterisk. We should keep the symbol in the param column.
- You use `photo` instead of `attachment` in some description.

The others are just nitpicking.

**Pull request merged**